source/coap_connection_handler.c

```diff
@@ -235,6 +235,12 @@ static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephem
         socket_setsockopt(this->listen_socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_MULTICAST_HOPS, &(const int16_t) {
             16
         }, sizeof(int16_t));
+
+        // Set socket option to receive packet info
+        socket_setsockopt(this->listen_socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_RECVPKTINFO, &(const bool) {
```

Comment:
Weird formatting.Put the compound literal on one line. Guess this is copying the weird astyle auto-formatting above - can you fix that too?